src/backend/joanie/core/models/activity_logs.py

```diff
@@ -88,55 +89,38 @@ class Meta:
         verbose_name_plural = _("activity_logs")
 
     @classmethod
-    def create_payment_succeeded_activity_log(cls, order, is_batch=False):
+    def create_payment_succeeded_activity_log(cls, order):
         """
         Create a payment succeeded activity log
         """
-        context = (
-            {"order_id": str(order.id)}
-            if not is_batch
-            else {"batch_order_id": str(order.id)}
-        )
         return cls.objects.create(
             user=order.owner,
             level=enums.ACTIVITY_LOG_LEVEL_SUCCESS,
-            context=context,
+            context={f"{order._meta.model_name}_id": str(order.id)},  # noqa: SLF001
             type=enums.ACTIVITY_LOG_TYPE_PAYMENT_SUCCEEDED,
         )
 
     @classmethod
-    def create_payment_failed_activity_log(cls, order, is_batch=False):
+    def create_payment_failed_activity_log(cls, order):
```

Comment:
Here `order` alone is ambiguous as it can be a `BatchOrder` or an `Order`. IMO python typing can be helpful in this case.
```suggestion
    def create_payment_failed_activity_log(cls, instance: Order | BatchOrder):
```